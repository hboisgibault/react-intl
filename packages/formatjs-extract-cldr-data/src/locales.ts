/*
 * Copyright 2015, Yahoo Inc.
 * Copyrights licensed under the New BSD License.
 * See the accompanying LICENSE file for terms.
 */
// These are the exceptions to the default algorithm for determining a locale's
// parent locale.
import * as AVAILABLE_LOCALES from 'cldr-core/availableLocales.json';

export function getAllLocales() {
  return AVAILABLE_LOCALES.availableLocales.full.sort();
}

export function getAllLanguages() {
  return new Set(
    AVAILABLE_LOCALES.availableLocales.full.map(l => l.split('-')[0])
  );
}
