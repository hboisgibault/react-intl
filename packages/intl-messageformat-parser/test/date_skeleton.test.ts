import {pegParse} from '../src/parser';
import {printAST} from '../src/printer';
import {parseDateTimeSkeleton} from '../src/skeleton';

test.each([
  `yyyy.MM.dd G 'at' HH:mm:ss vvvv`,
  `EEE, MMM d, ''yy`,
  `h:mm a`,
  ``,
])('case: %p', skeleton => {
  const ast = pegParse(`{0, date, ::${skeleton}}`);
  expect(ast).toMatchSnapshot();
  expect(printAST(ast)).toMatchSnapshot();
});

test.each([
  `yyyy.MM.dd G 'at' HH:mm:ss zzzz`,
  `EEE, MMM d, ''yy`,
  `h:mm a`,
  ``,
])('case: %p', skeleton => {
  expect(parseDateTimeSkeleton(skeleton)).toMatchSnapshot();
});
