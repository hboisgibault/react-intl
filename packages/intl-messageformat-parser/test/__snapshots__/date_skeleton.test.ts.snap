// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`case: "" 1`] = `
Array [
  Object {
    "style": "::",
    "type": 3,
    "value": "0",
  },
]
`;

exports[`case: "" 2`] = `"{0, date, ::}"`;

exports[`case: "" 3`] = `Object {}`;

exports[`case: "EEE, MMM d, ''yy" 1`] = `
Array [
  Object {
    "style": Object {
      "pattern": "EEE, MMM d, ''yy",
      "type": 1,
    },
    "type": 3,
    "value": "0",
  },
]
`;

exports[`case: "EEE, MMM d, ''yy" 2`] = `"{0, date, ::EEE, MMM d, ''yy}"`;

exports[`case: "EEE, MMM d, ''yy" 3`] = `
Object {
  "day": "numeric",
  "month": "short",
  "weekday": "short",
  "year": "2-digit",
}
`;

exports[`case: "h:mm a" 1`] = `
Array [
  Object {
    "style": Object {
      "pattern": "h:mm a",
      "type": 1,
    },
    "type": 3,
    "value": "0",
  },
]
`;

exports[`case: "h:mm a" 2`] = `"{0, date, ::h:mm a}"`;

exports[`case: "h:mm a" 3`] = `
Object {
  "hour": "numeric",
  "hour12": true,
  "hourCycle": "h12",
  "minute": "2-digit",
}
`;

exports[`case: "yyyy.MM.dd G 'at' HH:mm:ss vvvv" 1`] = `
Array [
  Object {
    "style": Object {
      "pattern": "yyyy.MM.dd G 'at' HH:mm:ss vvvv",
      "type": 1,
    },
    "type": 3,
    "value": "0",
  },
]
`;

exports[`case: "yyyy.MM.dd G 'at' HH:mm:ss vvvv" 2`] = `"{0, date, ::yyyy.MM.dd G 'at' HH:mm:ss vvvv}"`;

exports[`case: "yyyy.MM.dd G 'at' HH:mm:ss zzzz" 1`] = `
Object {
  "day": "2-digit",
  "era": "short",
  "hour": "2-digit",
  "hourCycle": "h23",
  "minute": "2-digit",
  "month": "2-digit",
  "second": "2-digit",
  "timeZoneName": "long",
  "year": "numeric",
}
`;
